python/GafferSceneUI/ShaderUI.py

```diff
@@ -364,3 +368,378 @@ def __graphEditorPlugContextMenu( graphEditor, plug, menuDefinition ) :
 	)
 
 GafferUI.GraphEditor.plugContextMenuSignal().connect( __graphEditorPlugContextMenu, scoped = False )
+
+
+##########################################################################
+# ShaderParameterDialog
+##########################################################################
+
+# A Path pointing to a shader or shader parameter belonging to one or more
+# IECore::ShaderNetworks. Shaders with identical names are merged
+# into a single location.
+
+class _DuplicateIconColumn ( GafferUI.PathColumn ) :
+
+	def __init__( self, title, property ) :
+
+		GafferUI.PathColumn.__init__( self )
+
+		self.__title = title
+		self.__property = property
+
+	def cellData( self, path, canceller ) :
+
+		data = self.CellData( path.property( self.__property ) )
+
+		if "shader:instances" in path.propertyNames() and path.property( "shader:instances" ) > 1 :
+			data.icon = "duplicate.png"
+			data.toolTip = (
+				("Shader" if len( path ) == 1 else "Parameter" ) +
+				" occurs in multiple " +
+				( "networks." if len( path ) == 1 else "shaders." )
+			)
+
+		return data
+
+	def headerData( self, canceller ) :
+
+		return self.CellData( self.__title )
+
+class _LinkIconColumn ( GafferUI.PathColumn ) :
+
+	def __init__( self, title, property ) :
+
+		GafferUI.PathColumn.__init__( self )
+
+		self.__title = title
+		self.__property = property
+
+	def cellData( self, path, canceller ) :
+
+		data = self.CellData( path.property( self.__property ) )
+
+		if "shader:input" in path.propertyNames() :
+			input = path.property( "shader:input" )
+			if input is not None and input != "---" :
+				data.icon = "navigationArrow.png"
+				data.toolTip = "Scroll to parameter input."
+
+		return data
+
+	def headerData( self, canceller ) :
+
+		return self.CellData( "" )
```

Comment:
Seems like it'd be nice to have an "Input" heading for this.

Reply:
Fixed in a33b24bcc405410b18566d398ee70de50a725a96